thoth/report_processing/components/adviser.py

```diff
@@ -346,13 +407,12 @@ def _aggregate_data_per_interval(adviser_type_dataframe: pd.DataFrame, number_da
                 (adviser_type_dataframe["date"] >= low) & (adviser_type_dataframe["date"] <= high)
             ]
 
-            for index, row in subset_df[["jm_hash_id_encoded", "message", "date"]].iterrows():
-                encoded_id = row["jm_hash_id_encoded"]
-                if encoded_id not in aggregated_data[high].keys():
-                    aggregated_data[high][encoded_id] = {
-                        "jm_hash_id_encoded": f"type-{encoded_id}",
+            for index, row in subset_df[["message", "date"]].iterrows():
+                message = row["message"]
+                if message not in aggregated_data[high].keys():
+                    aggregated_data[high][message] = {
                         "message": row["message"],
```

Comment:
I think that `.iterrows()` has some bad  complexity issues (quadratic i guess?), maybe checkout https://medium.com/@rtjeannier/pandas-101-cont-9d061cb73bfc for more information (the article includes the time complexity comparisons)

Using a better approach with `loc` and `iloc` would probably look like this:
```suggestion
            for i in subset_df[["message", "date"]].index:
                message = df.loc[i, "message"]
                if message not in aggregated_data[high].keys():
                    aggregated_data[high][message] = {
                        "message": message,
```

Reply:
Thanks man!! On the long run might be good indeed :)

Reply:
Aight!
And btw
> (quadratic i guess?)

i meant cubic :D

Reply:
> > (quadratic i guess?)
> 
> i meant cubic :D

Something like [bogorows](https://en.wikipedia.org/wiki/Bogosort) would work here as well I guess.